src/github.com/getlantern/enproxy/proxy.go

```diff
@@ -169,11 +175,15 @@ func (p *Proxy) handleWrite(resp http.ResponseWriter, req *http.Request, lc *laz
 		badGateway(resp, fmt.Sprintf("Unable to write to connOut: %s", err))
 		return
 	}
-	if p.Host != "" {
-		// Always feed this so clients will be guaranteed to reach
-		// this particular proxy even if they originally reached us
-		// through (e.g.) DNS round robin.
-		resp.Header().Set(X_ENPROXY_PROXY_HOST, p.Host)
+	host := ""
+	if p.HostFn != nil {
+		host = p.HostFn(req)
+	}
+	if host == "" {
+		host = p.Host
+	}
+	if host != "" {
```

Comment:
The purpose of HostFn (and Host) is documented where the fields are defined, but I'll add this comment back since it can't hurt, I guess.

The Host is only kept around for backwards compatibility, since enproxy is a library we may expect other people to use.  I can't see why anyone would use both Host *and* HostFn.  HostFn is given preference because if you use that the only reason I can think about for providing a Host too is if you've forgotten to remove it.  I'll add some comments to this effect too.